src/moi_wrapper.jl

```diff
@@ -21,19 +21,34 @@ julia> x = model.add_variable(model)
 julia> model.add_constraint(model, ...)
 ```
 """
-function diff_optimizer(optimizer_constructor)::Optimizer
-    optimizer =
-        MOI.instantiate(optimizer_constructor; with_bridge_type = Float64)
+function diff_optimizer(
+    optimizer_constructor;
+    method = nothing,
+    with_parametric_opt_interface::Bool = false,
+    with_bridge_type = Float64,
+    with_cache::Bool = true,
+)
+    optimizer = MOI.instantiate(optimizer_constructor; with_bridge_type)
     # When we do `MOI.copy_to(diff, optimizer)` we need to efficiently `MOI.get`
     # the model information from `optimizer`. However, 1) `optimizer` may not
     # implement some getters or it may be inefficient and 2) the getters may be
     # unimplemented or inefficient through some bridges.
     # For this reason we add a cache layer, the same cache JuMP adds.
-    caching_opt = MOI.Utilities.CachingOptimizer(
-        MOI.Utilities.UniversalFallback(MOI.Utilities.Model{Float64}()),
-        optimizer,
-    )
-    return Optimizer(caching_opt)
+    caching_opt = if with_cache
+        MOI.Utilities.CachingOptimizer(
+            MOI.Utilities.UniversalFallback(
+                MOI.Utilities.Model{with_bridge_type}(),
+            ),
+            optimizer,
+        )
+    else
+        optimizer
+    end
+    if with_parametric_opt_interface
+        return POI.Optimizer(Optimizer(caching_opt; method = method))
```

Comment:
Adding keyword arguments to the `Optimizer` is what we've been moving away from, especially since it's not needed for the constructor, just use `MOI.set` after this `if`-`else`